plugins/common/socket/datagram.go

```diff
@@ -28,7 +28,7 @@ type packetListener struct {
 	Log                  telegraf.Logger
 
 	conn      net.PacketConn
-	decoder   internal.ContentDecoder
+	decoders  []internal.ContentDecoder
```

Comment:
How about using a [pool](https://pkg.go.dev/sync#Pool) instead?

Reply:
I didn't know about those, yes I'll use that instead!